core/lib/ccm-star.c

```diff
@@ -144,13 +140,38 @@ ctr(uint8_t *m, uint8_t m_len, const uint8_t* nonce)
   }
 }
 /*---------------------------------------------------------------------------*/
-static void set_key(const uint8_t *key) {
-    AES_128.set_key((uint8_t*)key);
+static void
+set_key(const uint8_t *key)
+{
+  AES_128.set_key(key);
+}
+/*---------------------------------------------------------------------------*/
+void
+aead(const uint8_t* nonce,
+    uint8_t* m, uint8_t m_len,
+    const uint8_t* a, uint8_t a_len,
+    uint8_t *result, uint8_t mic_len,
+    int forward)
+{
+  if(!forward) {
+    /* decrypt */
+    ctr(nonce, m, m_len);
+  }
+  
+  mic(nonce,
```

Comment:
Maybe the multiple line breaks aren't necessary in this function call since the identifiers are short.